docs/sdk/src/reference_docs/omni_node.rs

```diff
@@ -0,0 +1,170 @@
+//! * Next, `chain-spec-builder` is used to generate a `chainspec.json`, either for development, or
+//!   for production. References:
+//! 	* [`crate::reference_docs::chain_spec_genesis`]
+//! * For local development, the following options are available:
+//! 	* `polkadot-omni-node` (notably, with [`--dev-block-time`]). References:
+//! 		* [`crate::guides::your_first_node`]
+//! 	* External tools such as `chopsticks`, `zombienet`.
+//! 		* See the `README.md` file of the `polkadot-sdk-parachain-template`.
+//! * For production `polkadot-omni-node` can be used out of the box.
+//! * For further customization [`polkadot_omni_node_lib`] can be used.
+//!
+//! ## Appendix
+//!
+//! ### Consensus Engine
+//!
+//! This section describes how the interdependence between the node and the runtime is related to
+//! the consensus engine.
+//!
+//! In any given substrate-based chain, both the node and the runtime will have their own
+//! opinion/information about what consensus engine is going to be used.
+//!
+//! In practice, the majority of the implementation of any consensus engine is in the node side, but
+//! the runtime also typically needs to expose a custom runtime-api to enable the particular
+//! consensus engine to work, and that particular runtime-api is implemented by a pallet
+//! corresponding to that consensus engine.
+//!
+//! For example, taking a snippet from [`solochain-template-runtime`], the runtime has to provide
+//! this additional runtime-api (compared to [`minimal-template-runtime`]), if the node software is
+//! configured to use the Aura consensus engine:
+//!
+//! ```ignore
+//! impl sp_consensus_aura::AuraApi<Block, AuraId> for Runtime {
+//!     fn slot_duration() -> sp_consensus_aura::SlotDuration {
+//!         ...
+//!     }
+//!     fn authorities() -> Vec<AuraId> {
+//!         ...
+//!     }
+//! }
+//! ````
+//!
+//! For simplicity, we can break down "consensus" into two main parts:
+//!
+//! * Block Authoring: Deciding who gets to produce the next block.
+//! * Finality: Deciding when a block is considered final.
+//!
+//! For block authoring, there are a number of options:
+//!
+//! * [`sc_consensus_manual_seal`]: Useful for testing, where any node can produce a block at any
+//!   time. This is often combined with a fixed interval at which a block is produced.
+//! * [`sc_consensus_aura`]/[`pallet_aura`]: A simple round-robin block authoring mechanism.
+//! * [`sc_consensus_babe`]/[`pallet_babe`]: A more advanced block authoring mechanism, capable of
+//!   anonymizing the next block author.
+//! * [`sc_consensus_pow`]: Proof of Work block authoring.
+//!
+//! For finality, there is one main option shipped with polkadot-sdk:
+//!
+//! * [`sc_consensus_grandpa`]/[`pallet_grandpa`]: A finality gadget that uses a voting mechanism to
+//!   decide when a block
+//!
+//! **The most important lesson here is that the node and the runtime must have matching consensus
```

Comment:
How omni-node addresses this?

Reply:
It is not really addressed by omni-node, it is a data point that is important to know, therefore added under "Appendix". 

I will make it more clear and actionable.